calculator.rb

```diff
@@ -0,0 +1,148 @@
+# including the length of the array (which should be = 3 after parsing),
+# and the format of the 1st and 3rd elements of the array, which should be numeric.
+def validation(parsed_array)
+  if parsed_array.length > 3
+    puts "Sorry, this calculator can't handle operations that complex."
+    # If the array length is greater than 3, the user likely entered more than one operator. Returns nil.
+  elsif parsed_array.length < 3
+    puts "Hmm. It doesn't look like the operation you've entered is valid for this calculator."
+    # If the array length is less than 3, the user likely didn't enter enough information,
+    # or didn't enter it in a manner that the parser could handle. Returns nil.
+  elsif (parsed_array[0] =~ /^[-+]?\d+(\.\d+)?$/).nil? || (parsed_array[2] =~ /^[-+]?\d+(\.\d+)?$/).nil?
+    puts "Hmm. It doesn't look like the operation you've entered is valid for this calculator."
+    # This checks if the 1st and 3rd elements of the array match a regex that
+    # permits an optional +/- sign at the start of string, digits 0-9, and optional digits 0-9 after the decimal point.
+    # Returns nil.
+  else
+    return parsed_array
+    # If none of the above applies, returns the parsed array unmodified.
+  end
+end
+
+# This method checks if the number argument (in string format) includes a decimal point.
+# Returns the string as a float if it includes a decimal point and an integer if it does not.
+def integer_or_float(x)
+  if x.include?(".")
+    return x.to_f
+  else
+    return x.to_i
+  end
+end
+
+# Addition method
+def add(x, y)
+  return x + y
+end
+
+# Subtraction method
+def subtract(x, y)
+  return x - y
+end
+
+# Multiplication method
+def multiply(x, y)
+  return x * y
+end
+
+# Division method. If the divisor is 0, returns string notifiying user of their error.
+# If both arguments are integers and not divisible, converts the dividend to a float and returns the result of the float calculation.
+# Otherwise, returns the result of the integer or float calculation, depending on the data types of the arguments.
+def divide(x, y)
+  if y == 0
+    return "Psst. You can't divide by zero. Please try again."
+  elsif x.is_a?(Integer) && y.is_a?(Integer) && x % y != 0
+    return x.to_f / y
+  else
+    return x / y
+  end
+end
+
+# Exponent method
+def exponent(x, y)
+  return x ** y
+end
+
+# Modulo method. If the 2nd argument is 0, returns string notifying user of their error.
+def modulo(x, y)
+  if y == 0
+    return "Psst. You can't divide by zero. Please try again."
+  else
+    return x % y
+  end
+end
+
+# Informs user of the limits of the calculator, the possible operators, and prompts user for their calculation.
+puts "Welcome to my (very limited) calculator. This calculator can perform basic arithmetic operations between two numbers."
+puts "\nThe accepted operators are: +, -, *, /, ^, or %, or the words add, subtract, multiply, divide, exponent, or modulo."
```

Comment:
I would suggest an example of valid input.  At first I was entering just the operator (-,+,/,*, etc).